sled-agent/src/sled_agent.rs

```diff
@@ -94,6 +97,20 @@ pub enum Error {
 
     #[error("Error resolving DNS name: {0}")]
     ResolveError(#[from] internal_dns_client::multiclient::ResolveError),
+
+    #[error("I/O Error accessing file {path}: {err}")]
+    Io { path: PathBuf, err: std::io::Error },
+
+    #[error(
+        "Unexpected object in dladm cache file which cannot be deleted automatically
+        This is related to https://github.com/oxidecomputer/omicron/issues/1679 ; ideally
+        We'd never leave around this half-deleted objects, and we'd automatically remove
+        them if we ever saw them.
```

Comment:
I feel the second half, `ideally, ...` is not really necessary in this message. I think what would be more helpful is an explicit statement that the below steps _must_ by the developer to get around this. Something like: `To fix this, you must do the following:`.

Reply:
Simplified error message